app/src/main/java/com/example/nvdovin/weatherapp/presentation/application/dagger/AppComponent.java

```diff
@@ -2,14 +2,18 @@
 
 import android.content.Context;
 
+import com.example.nvdovin.weatherapp.data.dao.DaoSession;
+
 import javax.inject.Singleton;
 
 import dagger.Component;
 
-@Component(modules = {AppModule.class})
+@Component(modules = {AppModule.class, DatabaseModule.class})
 @Singleton
 public interface AppComponent {
 
     Context context();
 
+    DaoSession daoSession();
```

Comment:
If you create Data Access Services you don't need to have the DaoSession here, you will have here just the Database Services for the entities you have.